tests/testthat/test-misc_cpp.R

```diff
@@ -3,7 +3,7 @@ context("Testing misc C++ functions")
 test_that(
   "C++ factorial function is correct",
   expect_equal(
-    lapply(1:10, BayesMallows:::factorial),
+    lapply(1:10, factorial),
```

Comment:
Looks fine here, but we gotta be careful not to create conflicts with `Rcpp::factorial()`. As a matter of fact, why do we need own own version of it anyway?

Reply:
Thanks! I wasn't aware of that Rcpp function. I'll get rid of our own in a new commit to this PR.

Reply:
Looks like `divide_by_fact()` is the only usage of `Rcpp::factorial()`in the package, and `BayesMallows:::factorial()` also has one single use on `hamming_logz()`.

Reply:
An issue with `Rcpp::factorial` is that it takes a `Rcpp::NumericVector` and returns a `Rcpp::NumericVector`, thus requiring type conversion. I'll try to use the `tgamma` function from `math.h` instead, for the cases where `BayesMallows:::factorial` was used previously. BRB.

Reply:
Nice if you could have a look at the latest change @wleoncio. Found a factorial function in boost which seems to work in both cases. I'll try to use boost for the binomial coefficient as well, but that comes in a later PR.

Reply:
Ooops, that introduced another dependency, which worked locally on my computer but not elsewhere. Reverting now.